src/tests/ftest/util/mdtest_test_base.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/python
+'''
+  (C) Copyright 2019 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+'''
+from __future__ import print_function
+
+import os
+
+from apricot import TestWithServers
+from daos_api import DaosPool
+from mpio_utils import MpioUtils
+from mdtest_utils import MdtestCommand, MdtestFailed
+
+class MdtestBase(TestWithServers):
+    """
+    Base mdtest class
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a MdtestBase object."""
+        super(MdtestBase, self).__init__(*args, **kwargs)
+        self.mdtest_cmd = None
+        self.processes = None
+        self.hostfile_clients_slots = None
+
+    def setUp(self):
+        """Set up each test case."""
+        # Start the servers and agents
+        super(MdtestBase, self).setUp()
+
+        # Get the parameters for Mdtest
+        self.mdtest_cmd = MdtestCommand()
+        self.mdtest_cmd.get_params(self)
+        self.processes = self.params.get("np", '/run/mdtest/client_processes/*')
+        self.manager = self.params.get("manager", '/run/mdtest/*', "MPICH")
+
+    def tearDown(self):
+        """Tear down each test case."""
+        try:
+            if self.pool is not None:
+                self.pool.destroy(1)
+        finally:
+            # Stop the servers and agents
+            super(MdtestBase, self).tearDown()
+
+    def execute_mdtest(self):
+        """
+        Execute mdtest with optional overrides for mdtest flags
+        and object_class.
+        If specified the mdtest flags and mdtest daos object class parameters
+        will override the values read from the yaml file.
+        Args:
+            mdtest_flags (str, optional): mdtest flags. Defaults to None.
+            object_class (str, optional): daos object class. Defaults to None.
+        """
+        # Get the parameters used to create a pool
+        mode = self.params.get("mode", "/run/pool/*")
```

Comment:
Lines 77-89 could be replaced by the following:
```
self.pool = TestPool(self.context, self.log)
self.pool.get_params(self)

# Create a pool
self.pool.create()
```

The only additional change is renaming `setname` to `group` in the yaml file